libraries/standard/http/include/http_client.h

```diff
@@ -1,347 +1,371 @@
 #ifndef HTTP_CLIENT_H_
 #define HTTP_CLIENT_H_
 
-#include <stdint.h>
-#include <stdbool.h>
-
 /**
  * @brief Maximum size, in bytes, of headers allowed from the server.
- * 
+ *
  * If the total size in bytes of the headers sent from this server exceeds this
  * configuration, then the status code #HTTP_SECURITY_ALERT_HEADERS is
  * returned from #HTTPClient_Send.
  */
 #ifndef HTTP_MAX_RESPONSE_HEADERS_SIZE_BYTES
-    #define HTTP_MAX_RESPONSE_HEADERS_SIZE_BYTES      2048U
+    #define HTTP_MAX_RESPONSE_HEADERS_SIZE_BYTES    2048U
 #endif
 
 /**
  * @brief The HTTP header "User-Agent" value.
- * 
+ *
  * The following headerline is automatically written to
  * #HTTPRequestHeaders_t.pBuffer:
  * "User-Agent: my-platform-name\r\n"
- */ 
+ */
 #ifndef HTTP_USER_AGENT_VALUE
-    #define HTTP_USER_AGENT_VALUE       "my-platform-name"
+    #define HTTP_USER_AGENT_VALUE    "my-platform-name"
 #endif
 
 /**
- * Supported HTTP request methods.
+ * @brief Supported HTTP request methods.
  */
-#define HTTP_METHOD_GET                 "GET"  /**< HTTP Method GET. */
-#define HTTP_METHOD_PUT                 "PUT"  /**< HTTP Method PUT. */
-#define HTTP_METHOD_POST                "POST" /**< HTTP Method POST. */
-#define HTTP_METHOD_HEAD                "HEAD" /**< HTTP Method HEAD. */
+#define HTTP_METHOD_GET          "GET"             /**< HTTP Method GET. */
+#define HTTP_METHOD_PUT          "PUT"             /**< HTTP Method PUT. */
+#define HTTP_METHOD_POST         "POST"            /**< HTTP Method POST. */
+#define HTTP_METHOD_HEAD         "HEAD"            /**< HTTP Method HEAD. */
+
+/**
+ * @brief The HTTP protocol version of this library is HTTP/1.1.
+ */
+#define HTTP_PROTOCOL_VERSION    "HTTP/1.1"
```

Comment:
This can be in the source file instead

Reply:
Oh but user might wanna define different HTTP_PROTOCOL_VERSION?

Reply:
Users will not be allowed to change the protocol version. This should go into the internal.h header.